### PR TITLE
[CWS] Add kernel space cgroup selector to trigger activity dumps

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR introduces a new feature to automatically trigger activity dumps based on a kernel space cgroup selector mechanism. The goal is to progressively generate activity dumps for all the cgroups on a host, in a configurable manner. The following configuration parameters were added:
- `runtime_security_config.activity_dump.tags_resolution_period`: defines the period at which the activity dump manager should try to resolve missing container tags.
- `runtime_security_config.activity_dump.traced_cgroups_count`: defines the maximum count of cgroups that should be monitored concurrently. Set this parameter to -1 to monitor all cgroups at the same time. Leave this parameter to 0 to prevent the generation of activity dumps based on cgroups.
- `runtime_security_config.activity_dump.traced_event_types`: defines the list of events that should be captured in an activity dump. Leave this parameter empty to monitor all event types. If not already present, the `exec` event will automatically be added to this list.
- `runtime_security_config.activity_dump.default_dump_timeout`: defines the default activity dump timeout.
- `runtime_security_config.activity_dump.cgroups_wait_list_size`: defines the size of the cgroup wait list. The wait list is used to introduce a delay between 2 activity dumps of the same cgroup.

The `tags` selector was also removed from the `security-agent` command line because it would have conflicted with the cgroup selector (plus the feature wasn't really working because of the container tags resolution delay).

The following metrics were also added:
- `datadog.runtime_security.activity_dump.size_in_bytes`: the name of the metric used to report the size of the generated activity dumps in bytes.
- `datadog.runtime_security.activity_dump.active_dumps`: the name of the metric used to report the number of active dumps.

### Motivation

The goal of the PR is to introduce a way to trigger activity dumps without user interaction.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — APPROVED** [review]
`pkg/config/config.go` 👍🏽

File: pkg/security/probe/activity_dump.go [hunk 1]
```diff
@@ -201,11 +247,20 @@ func (ad *ActivityDump) dump() {
...
 	if err != nil {
 		seclog.Errorf("couldn't write ActivityDump: %s\n", err)
 		return
 	}
+
+	// send dump size
+	if n > 0 {
+		var tags []string
+		if err = ad.adm.probe.statsdClient.Gauge(metrics.MetricActivityDumpSizeInBytes, float64(n), tags, 1.0); err != nil {
+			seclog.Warnf("couldn't send %s metric: %w", metrics.MetricActivityDumpSizeInBytes, err)
```
Comment (paulcacheux) [inline pkg/security/probe/activity_dump.go:260]:
```suggestion
			seclog.Warnf("couldn't send %s metric: %v", metrics.MetricActivityDumpSizeInBytes, err)
```
Warnf doesn't support %w

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/probe/activity_dump.go (hunk 1 above)
Comment (Gui774ume) [reply pkg/security/probe/activity_dump.go]: 👍🏻

File: pkg/config/config.go [hunk 2]
```diff
@@ -996,6 +996,12 @@ func InitConfig(config Config) {
 	config.BindEnv("runtime_security_config.enable_runtime_compiled_constants")
 	config.BindEnvAndSetDefault("runtime_security_config.activity_dump_manager.enabled", false)
 	config.BindEnvAndSetDefault("runtime_security_config.activity_dump_manager.cleanup_period", 30)
+	config.BindEnvAndSetDefault("runtime_security_config.activity_dump_manager.tags_resolution_period", 60)
```
Comment (lebauce) [inline pkg/config/config.go:999]: nit: activity_dump instead of activity_dump_manager (sounds a bit too implementation level)

Comment (Gui774ume) [reply pkg/config/config.go]: 👍🏻

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**